lexer.py

```diff
@@ -0,0 +1,127 @@
+from tokens import TokenTypes, Token
+
+###############################################################################
+#                                                                             #
+#   LEXER                                                                     #
+#                                                                             #
+###############################################################################
+
+ESCCHARS = {
+    "n":  "\n",
+    "t":  "\t",
+    "\\": "\\",
+    "'": "'",
+    "r":  "\r",
+}
+
+RESERVED_WORDS = {
+            "SELECT",
+            "FROM",
+            "EDGES",
+            "NODES",
+            "TRAVERSE",
+            "BY",
+            "DEPTH",
+            "BREADTH",
+            "WHERE",
+            "LIMIT",
+}
+
+class Lexer(object):
+    def __init__(self, text):
+        self.text = text         # string input
+        self.pos = 0             # self.pos is index into self.text
+        self.line = 1            # current line
+        self.current_char = self.text[0] # current char
+        self.tokens = []         # tokens so far
+    def _error(self, msg=""):
+        raise Exception("LexingError, line %s: " + msg + "." % self.line)
+    def _peek(self, lookahead=1):
+        _peek_pos = self.pos + lookahead
+        if _peek_pos >= len(self.text):
+            return None
+        else:
+            return self.text[_peek_pos]
+    def _advance(self):
+        self.pos += 1
+        if self.pos < len(self.text):
+            self.current_char = self.text[self.pos]
+            if self.current_char == "\n":
+                self.line += 1
+        else:
+            self.current_char = None
+    def _eat_white_space(self):
+        while self.current_char == " ": self._advance()
+    def _add_token1(self, token_type):
```

Comment:
`_add_token1` and `_add_token2` are not very descriptive names.

Reply:
`_add_one_char_token` and `_add_two_char_token`, maybe? Or maybe as a suffix: `_add_token_one_char` and `_add_token_two_char`.

I'll also accept keeping the current names if you add a short docstring to both methods.